src/components/useIncrementalResponse.ts

```diff
@@ -0,0 +1,77 @@
+import { useCallback, useEffect, useRef, useState } from 'react'
+import IncrementalResponse from 'src/libs/ajax/IncrementalResponse'
+import LoadedState, { NoneState } from 'src/libs/type-utils/LoadedState'
+
+
+type GetIncrementalResponse<T> = (options: { signal: AbortSignal }) => Promise<IncrementalResponse<T>>
+
+type UseIncrementalResponseResult<T> = {
+  state: Exclude<LoadedState<T[]>, NoneState>
```

Comment:
@msilva-broad I got around my annoyance with having to check for `status === 'None'` with this.